portal-impl/src/com/liferay/portal/service/impl/WorkflowDefinitionLinkLocalServiceImpl.java

```diff
@@ -382,6 +397,33 @@ public WorkflowDefinitionLink updateWorkflowDefinitionLink(
 		return workflowDefinitionLinkPersistence.update(workflowDefinitionLink);
 	}
 
+	@Override
+	public WorkflowDefinitionLink updateWorkflowDefinitionLink(
+			String externalReferenceCode, long userId, long companyId,
+			long groupId, String className, long classPK, long typePK,
+			String workflowDefinitionName, int workflowDefinitionVersion)
+		throws PortalException {
+
+		WorkflowDefinitionLink serviceBuilderWorkflowDefinitionLink =
+			workflowDefinitionLinkPersistence.fetchByERC_G(
+				externalReferenceCode, groupId);
+
+		if (serviceBuilderWorkflowDefinitionLink != null) {
```

Comment:
It would be better if you had the `if (serviceBuilderWorkflowDefinitionLink == null)` condition here for the add flow and then leave the rest of the method with the update logic as we have in this pre-existing [method](https://github.com/liferay/liferay-portal/blob/9286cbb909d0eb4cf3480a7b52b4d9a98c3b438c/portal-impl/src/com/liferay/portal/service/impl/WorkflowDefinitionLinkLocalServiceImpl.java#L362-L382).